### PR TITLE
tests(smoke): convert core tests to single-expectations format

### PR DESCRIPTION
**Pull request by brendankenny**
part 2/2 of #11950. PR currently based on top of #12818

This splits up our core tests so there's a separate testDefn and ID for every smoke expectation. The splitting up is done here and tentative IDs are chosen, but I wanted some bike shedding input before I tried to put everything in the same format.

For reviewers:
- this is much, _much_ easier to review with [?w=1](https://github.com/GoogleChrome/lighthouse/pull/12819/files?w=1). Lots of indentation changes.
- Most important file is `core-tests.js`, as that's where the test IDs are set.
   (I'll be honest: there are a lot of these, and about halfway through this I stopped trying very hard with the names if the tests weren't super obvious how they were different from each other. I'm looking at you `redirects` and `csp` :P Please suggest better IDs than I have here)
- expectations are functionally all the same, but feel free to pick a smoke file you care about and double check it's still good and offer opinions on the changes in them.

For bikeshedders:
- IDs!
- #11950 suggested a file per expectation, but in some cases there are shared properties between expectations, and multiple named exports could maybe work ok. Do we want to go all the way and still do a file per expectation (and import shared stuff), or have a mixture of single and multiples, or...? Splitting into different files should be easy now, so I'm happy to go either way.
- We have grouped IDs where e.g. the old `errors` becomes `errors-infinite-loop`, `errors-expired-ssl` etc. Handy for running with just `yarn smoke errors`, but:
  - do we want to formalize a `category-specific` naming format?
  - are one-off tests ok? e.g. there's only one `oopif` tests, so it's still just called `oopif`. Is that ok or should we preemptively call it `oopif-requests` just in case one day we have `oopif-some-other-thing`?

### CODE REVIEW

File: lighthouse-cli/test/smokehouse/test-definitions/issues/mixed-content.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/**
```
Comment (brendankenny) [inline lighthouse-cli/test/smokehouse/test-definitions/issues/mixed-content.js:1]: moved from `dbw-expectations.js` as dbw is kind of its own thing and issues are worthy of their own smoke tests

File: lighthouse-cli/test/smokehouse/test-definitions/lantern/lantern-expectations.js [hunk 2]
```diff
@@ -6,120 +6,150 @@
...
-          // With the requestIdleCallback shim in place 1ms tasks should not block at all and should max add up to
-          // 12.5 ms each, which would result in 50ms under a 4x simulated throttling multiplier and therefore in 0 tbt
-          numericValue: '<=100',
-        },
+};
+
+/**
+ * @type {Smokehouse.ExpectedRunnerResult}
+ */
+const ricShort = {
```
Comment (patrickhulce) [inline lighthouse-cli/test/smokehouse/test-definitions/lantern/lantern-expectations.js:115]: idleCallbackShort?

`ric` lowercase is harder to remember what it is than `rIC`, probably should've done that with the `ric-shim` filename originally 😊

File: lighthouse-cli/test/smokehouse/test-definitions/lantern/lantern-expectations.js [hunk 3]
```diff
@@ -6,120 +6,150 @@
...
-          // sometimes Chrome's requestIdleCallback won't fire the full 40 if the machine is under load,
-          // so be generous with how much slack to give in the expectations.
-          numericValue: '>2500',
-        },
+};
+
+/**
+ * @type {Smokehouse.ExpectedRunnerResult}
+ */
+const ricLong = {
```
Comment (patrickhulce) [inline lighthouse-cli/test/smokehouse/test-definitions/lantern/lantern-expectations.js:132]: same

File: lighthouse-cli/test/smokehouse/test-definitions/redirects/expectations.js [hunk 4]
```diff
@@ -6,129 +6,151 @@
...
-      ],
+};
+
+// TODO: what's the describable difference between this and the above? Also, better names.
+/**
+ * @type {Smokehouse.ExpectedRunnerResult}
+ * Expected Lighthouse audit values for a site with a client-side redirect (2s + 5s),
+ * paints at 2s, then a server-side redirect (1s).
+ */
+const clientPaintServer2 = {
```
Comment (patrickhulce) [inline lighthouse-cli/test/smokehouse/test-definitions/redirects/expectations.js:135]:
```suggestion
/**
 * @type {Smokehouse.ExpectedRunnerResult}
 * Expected Lighthouse audit values for a site with a client-side redirect (2s + 5s),
 * paints at 2s, then a server-side redirect (1s), followed by a history.pushState to an unrelated URL.
 */
const historyPushState = {
```

File: lighthouse-cli/test/smokehouse/test-definitions/core-tests.js [hunk 5]
```diff
@@ -5,71 +5,217 @@
...
+  expectations: seo.failing,
+  config: require('./seo/seo-config.js'),
+}, {
+  id: 'seo-status-403',
+  expectations: seo.status403,
   config: require('./seo/seo-config.js'),
 }, {
-  id: 'offline',
-  expectations: require('./offline-local/offline-expectations.js'),
+  id: 'seo-tap-target',
```
Comment (patrickhulce) [inline lighthouse-cli/test/smokehouse/test-definitions/core-tests.js:106]:
```suggestion
  id: 'seo-tap-targets',
```

File: lighthouse-cli/test/smokehouse/test-definitions/core-tests.js [hunk 6]
```diff
@@ -5,71 +5,217 @@
...
+}, {
+  id: 'redirects-client-paint-server',
+  expectations: redirects.clientPaintServer,
+  config: require('./redirects/redirects-config.js'),
+}, {
+  id: 'redirects-single-client',
+  expectations: redirects.singleClient,
+  config: require('./redirects/redirects-config.js'),
+}, {
+  id: 'redirects-client-paint-server-2',
```
Comment (patrickhulce) [inline lighthouse-cli/test/smokehouse/test-definitions/core-tests.js:90]:
```suggestion
  id: 'redirects-history-pushstate',
```

**Review by patrickhulce — APPROVED** [review]
names mostly seem fine to me! 🎉 

thanks for trudging through this super mechanical drudgery :D

File: lighthouse-cli/test/smokehouse/test-definitions/redirects/expectations.js (hunk 4 above)
Comment (patrickhulce) [reply lighthouse-cli/test/smokehouse/test-definitions/redirects/expectations.js:135]: NOTE: will require a few other `historyPushState` renames as well

**Comment by paulirish** [comment]
yay death to pwa2 and pwa3! :) 

> Do we want to go all the way and still do a file per expectation (and import shared stuff)

yeah. i think doing that allows us to fix the file/folder structure of test-definitions which i'm really excited about. i think it makes sense as a follow up PR; the scope of this one is great.

> do we want to formalize a category-specific naming format?

eh, nah.

> are one-off tests ok? e.g. there's only one oopif tests, so it's still just called oopif. Is that ok or should we preemptively call it oopif-requests just in case one day we have oopif-some-other-thing?

yeah lets leave it alone. shorter is better, so lets only elongate when we have to.

i'm happy with these slugs (except for csp-base which should be csp-basics or something)

woo!

**Comment by brendankenny** [comment]
SG. We'll go with these (except for the CSP ones, @adamraine is about to take a look) but we have a second chance to discuss them in the next PR.

File: lighthouse-cli/test/smokehouse/test-definitions/csp/csp-expectations.js [hunk 7]
```diff
@@ -23,86 +23,99 @@ const blockAllExceptInlineScriptCsp = headersParam([[
...
-        sourceMapUrl: 'http://localhost:10200/source-map/script.js.map',
-        map: {},
-        errorMessage: undefined,
-      }],
+};
+
+/**
+ * @type {Smokehouse.ExpectedRunnerResult}
+ */
+const exceptInlineM92 = {
```
Comment (adamraine) [inline lighthouse-cli/test/smokehouse/test-definitions/csp/csp-expectations.js:92]:
```suggestion
const blockAll = {
```

File: lighthouse-cli/test/smokehouse/test-definitions/csp/csp-expectations.js [hunk 8]
```diff
@@ -23,86 +23,99 @@ const blockAllExceptInlineScriptCsp = headersParam([[
...
-        sourceMapUrl: 'http://localhost:10200/source-map/script.js.map',
-        map: {},
-        errorMessage: undefined,
-      }],
-    },
-    lhr: {
-      requestedUrl: 'http://localhost:10200/csp.html',
-      finalUrl: 'http://localhost:10200/csp.html',
-      audits: {},
+const base = {
```
[the remark this answers is not in the document]
Comment (adamraine) [reply lighthouse-cli/test/smokehouse/test-definitions/csp/csp-expectations.js:29]: Alternative: `allowAll`

**Comment by brendankenny** [comment]
looks great. Thanks!

File: lighthouse-cli/test/smokehouse/test-definitions/core-tests.js [hunk 9]
```diff
@@ -5,71 +5,217 @@
...
   expectations: require('./dobetterweb/dbw-expectations.js'),
   config: require('./dobetterweb/dbw-config.js'),
   runSerially: true, // Need access to network request assertions.
 }, {
-  id: 'redirects',
-  expectations: require('./redirects/expectations.js'),
+  id: 'issues-mixed-content',
+  expectations: require('./issues/mixed-content.js'),
+}, {
+  id: 'redirects-single-server',
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/test-definitions/core-tests.js:74]: how about going one step further and moving this entire object to what `expectations.js` exports? then this big chunk of code is just bringing it all together as an array.

Comment (brendankenny) [reply lighthouse-cli/test/smokehouse/test-definitions/core-tests.js:74]: Yeah, it's a little silly to have filenames for separate files and then IDs for them in a different file. That'll be in the above-mentioned (but undescribed, sorry) next PR.

**Pull request merged**